gateway/src/apicast/policy/upstream/policy.lua

```diff
@@ -0,0 +1,70 @@
+--- Upstream policy
+-- This policy allows to modify the host of a request based on its path.
+
+local resty_resolver = require('resty.resolver')
+local resty_url = require('resty.url')
+local format = string.format
+local ipairs = ipairs
+local match = ngx.re.match
+local table = table
+
+local _M = require('apicast.policy').new('Upstream policy')
+
+local new = _M.new
+
+local function change_upstream(url)
+  ngx.ctx.upstream = resty_resolver:instance():get_servers(
+    url.host, { port = url.port })
+
+  if url.path then
+    ngx.var.proxy_pass = format('%s://upstream%s?%s', url.scheme, url.path, ngx.var.args)
```

Comment:
This is going to always append `?`, right? Even though there are no args.

Reply:
I haved extracted this part of the code to its own method to make it clearer.